spec/controllers/tasks_controller_spec.rb

```diff
@@ -404,6 +404,48 @@
     end
   end
 
+  describe 'POST #duplicate' do
+    subject(:post_duplicate) { post :duplicate, params: {id: task.to_param} }
+
+    let(:group) { create(:group) }
+    let(:collection) { create(:collection) }
+
+    let!(:task) { create(:task, valid_attributes) }
+    let(:valid_attributes) do
+      {
+        user:,
+        title: 'title',
+        license: create(:license, name: 'license'),
+        groups: [group],
+        collections: [collection],
+      }
+    end
+
+    it 'creates a new Task' do
+      expect { post_duplicate }.to change(Task, :count).by(1)
+    end
+
+    it 'assigns a newly created task as @task' do
+      post_duplicate
+      expect(assigns(:task)).to be_persisted
+    end
+
+    it 'redirects to the created task' do
+      post_duplicate
+      expect(response).to redirect_to(Task.last)
+    end
+
+    it 'resets the groups' do
+      post_duplicate
+      expect(Task.last.groups).to eq([])
+    end
+
+    it 'resets the collections' do
+      post_duplicate
+      expect(Task.last.collections).to eq([])
+    end
```

Comment:
As discussed, I am not a fan of the `Task.last` statement (even though used already), since this will change with another primary key.

Reply:
It now implemented and is going to be uploaded alongside the other suggestions.